### DISCUSSION TITLE
How to update new Filters for v8 Pixijs?

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [pixijs/pixijs#11171] [Q&A]
I've got the following line of code:
`const brightnessFilter = new PIXI.Filter(undefined, this.toGreyShaderData, this.getGreyscaleFilter(currentImageObject.greyMode));`

I tried updating the code for v8 as below:
```
const brightnessFilter = new PIXI.Filter({
  glProgram: PIXI.GlProgram.from({
    vertex: undefined,
    fragment: this.toGreyShaderData
  }),
  resources: {
    timeUniforms: {
      ...this.getGreyscaleFilter(currentImageObject.greyMode)
    }
  },
});
```
But I'm getting the error:
Cannot read properties of undefined (reading 'substring')
Looking at the source it says the error occurs at ensurePrecision() function in pixi at `if (src.substring(0, 9) !== "precision") {`

**Comment by whyhw** [comment]
[src/filters/defaults/defaultFilter.vert](https://github.com/pixijs/pixijs/blob/be64cc099705f308f7a6bc26f6067d076091935c/src/filters/defaults/defaultFilter.vert)
Maybe you can use this code as the value of vertex.

**Reply by steven-twerdochlib** [reply]
So I tried this:
```
import { defaultFilterVertex } from '@pixi/core';

const brightnessFilter = new PIXI.Filter({
  glProgram: PIXI.GlProgram.from({
    vertex: defaultFilterVertex,
    fragment: this.toGreyShaderData
  }),
  resources: {
    timeUniforms: {
      ...this.getGreyscaleFilter(currentImageObject.greyMode)
    }
  },
});
```

But got this error: `TypeError: Cannot create property 'name' on number '0'` in the following function
```
 /**
   * Create a new Uniform group
   * @param uniformStructures - The structures of the uniform group
   * @param options - The optional parameters of this uniform group
   */
  constructor(uniformStructures, options) {
```

Note: I'm using vue3
